chain/network/src/network_protocol/tests.rs

```diff
@@ -170,3 +176,96 @@ fn serialize_deserialize() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn serialize_deserialize_ip_addr() {
+    let mut rng = make_rng(89028037453);
+
+    // Convert IpAddr to proto::IpAddr for protocol buffer serialization
+    let ip_addr_before_serialize: net::IpAddr = data::make_ipv4(&mut rng);
+    let proto_ip_addr_before_serialize: proto::IpAddr = proto::IpAddr::from(&ip_addr_before_serialize);
+    // Serialize and deserialize for transporting over the network
+    let ip_addr_bytes: Vec<u8> = proto_ip_addr_before_serialize.write_to_bytes().unwrap();
+    let proto_ip_addr_after_deserialize = proto::IpAddr::parse_from_bytes(&ip_addr_bytes).unwrap();
+    assert_eq!(proto_ip_addr_before_serialize,  proto_ip_addr_after_deserialize);
+    // Convert proto::IpAddr back to IpAddr to be independent of protocol buffer in near network
+    let result_deserialized_actual_ip_addr = net::IpAddr::try_from(&proto_ip_addr_after_deserialize);
+    assert!(result_deserialized_actual_ip_addr.is_ok());
+    let deserialized_actual_ip_addr = result_deserialized_actual_ip_addr.unwrap();
+    assert_eq!(ip_addr_before_serialize, deserialized_actual_ip_addr);
+}
+
+#[test]
+fn sign_and_verify_ip_addr_with_peer_id() {
+    // Design sign and verify algorithm for std::net::IpAddr
+    let seed = "123";
+    let node_key = SecretKey::from_seed(KeyType::ED25519, seed);
+    let peer_id = PeerId::new(node_key.public_key());
+    let mut rng = make_rng(89028037453);
+    let ip_addr: net::IpAddr = data::make_ipv4(&mut rng);
+    let ip_bytes: Vec<u8> = match ip_addr {
+        net::IpAddr::V4(ip) => ip.octets().to_vec(),
+        net::IpAddr::V6(ip) => ip.octets().to_vec(),
+    };
+    let signature = node_key.sign(&ip_bytes);
+    assert!(signature.verify(&ip_bytes, &node_key.public_key()));
+    assert!(signature.verify(&ip_bytes, peer_id.public_key()));
+
+    // Wrap ip address sign and verify algorithm with interfaces: OwnedIpAddress and SignedOwnedIpAddress
+    let owned_ip_address = OwnedIpAddress {
+        ip_address: ip_addr.clone(),
+    };
+    let signed_owned_ip_address: SignedOwnedIpAddress = owned_ip_address.sign(&node_key);
+    assert!(signed_owned_ip_address.verify(&node_key.public_key()));
+    assert!(signed_owned_ip_address.verify(peer_id.public_key()));
+}
+
+#[test]
+fn serialize_deserialize_owned_ip_address() {
```

Comment:
Redundant with other tests; also see recommendation not to make a special struct `OwnedIpAddress`.

Reply:
Removed the test but maintain the builder wrapper OwnedIpAddress as discussed offline